contracts/modules/grouping/GroupingModule.sol

```diff
@@ -183,13 +192,30 @@ contract GroupingModule is
     /// @param ipIds The IP IDs.
     function claimReward(address groupId, address token, address[] calldata ipIds) external whenNotPaused {
         IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupId));
-        // claim reward from group IPA's RoyaltyVault to group pool
-        pool.collectRoyalties(groupId, token);
         // trigger group pool to distribute rewards to group members vault
         uint256[] memory rewards = pool.distributeRewards(groupId, token, ipIds);
         emit ClaimedReward(groupId, token, ipIds, rewards);
     }
 
+    /// @notice Collects royalties into the pool, making them claimable by group member IPs.
+    /// @param groupId The address of the group.
+    /// @param token The address of the token.
+    /// @param snapshots The snapshot IDs of the royalty vault for the given group to collect royalties.
+    function collectRoyalties(
```

Comment:
Having the `collectRoyalties` function in the pool contract as it was before would save gas as it does not need to go through grouping module additional proxy call. We consider moving the function to pool contract if there is no upside of having it here.

Reply:
Thanks for the thoughtful inputs!
Agreed, moving the collectRoyalties function to the pool contract could save a proxy call. However, I believe it is still beneficial to keep it in the grouping module for the following reasons:

1. Transferring royalties to the pool is a common logic shared by all pools, so it makes sense to centralize this logic within the module rather than duplicating it across multiple pool contracts.
2. Having the function in the grouping module provides a unified entry point for all grouping-related functions, which improves consistency and maintainability.